inference-engine/src/mkldnn_plugin/nodes/psroi.cpp

```diff
@@ -42,250 +46,418 @@ class PSROIPoolingImpl: public ExtLayerBase {
+                                                     * partSize + partW] * transStd;
+                    const float transY = noTrans ? 0 :
+                                         bottomTrans[(((n * numClasses + classId) * 2 + 1) * partSize + partH)
+                                                     * partSize + partW] * transStd;
+
+                    const float wStart = w * binSizeW + roiStartW + transX * roiWidth;
+                    const float hStart = h * binSizeH + roiStartH + transY * roiHeight;
+
+                    float sum = 0;
+                    int count = 0;
+                    int gw = w * groupSize / pooledWidth;
+                    int gh = h * groupSize / pooledHeight;
+                    gw = (std::min)((std::max)(gw, 0), static_cast<int>(groupSize - 1));
+                    gh = (std::min)((std::max)(gh, 0), static_cast<int>(groupSize - 1));
+
+                    const inputType *offsetBottomData = srcData + (roiBatchInd * channels) * height * width;
+                    for (size_t ih = 0; ih < spatialBinsY; ih++) {
+                        for (size_t iw = 0; iw < spatialBinsX; iw++) {
+                            float w1 = wStart + iw * subBinSizeW;
+                            float h1 = hStart + ih * subBinSizeH;
+                            // bilinear interpolation
+                            if (w1 < -0.5 || w1 > width - 0.5 || h1 < -0.5 || h1 > height - 0.5)
+                                continue;
+                            w1 = static_cast<float>((std::min)((std::max)(static_cast<double>(w1), 0.0), width - 1.0));
+                            h1 = static_cast<float>((std::min)((std::max)(static_cast<double>(h1), 0.0), height - 1.0));
+                            int c1 = static_cast<int>((c * groupSize + gh) * groupSize + gw);
+                            float val = bilinearInterp<inputType>(offsetBottomData +
+                                                                  c1 * height * width, w1, h1, width,
+                                                                  inBlockSize, c1 % inBlockSize);
+
+                            sum += val;
+                            count++;
+                        }
+                    }
+                    dstData[dstIndex] = count == 0 ? 0 : sum / count;
+                });
             }
         });
 
-        for (int n = real_rois; n < nn; n++) {
+        for (int n = realRois; n < nn; n++) {
             parallel_for3d(nc, nh, nw, [&](int c, int h, int w) {
+                // we don't care about blocked format, because data is filled per batch
                 int index = n * nc * nh * nw + c * nh * nw + h * nw + w;
-                dst_data[index] = 0.0f;
+                dstData[index] = 0;
             });
         }
+    }
+
+    StatusCode execute(std::vector<Blob::Ptr>& inputs, std::vector<Blob::Ptr>& outputs, ResponseDesc *resp) noexcept override {
```

Comment:
Since the `execute` marked as `noexcept` ,  we must be sure that code inside the function does not throw exceptions. If we are not sure, it is better to wrap the executed code into a `try catch(...)` block to ensure that the function does not throw exceptions.

Reply:
Done